### DISCUSSION TITLE
Are there any docs available?

### DISCUSSION THREAD

**Discussion by ovizii** [mayneyao/eidos#119] [General]
I'm currently testing your instance at eidos.space, and I am a bit lost. You could generate some default docs, which are inside a folder called onboarding and add some docs for new people explaining the basic concept of your app like what is a space, how to use files/folders and how to structure stuff. Just some quick random ideas.

And btw. if you'd like to keep this kind of questions out of your issues, you could enable discussions for this project, jsut throwing that out there.

I've completed all settings, told it to save locally, gave permissions and yet eidos.space says "last backup unknown"… 

I am also wondering why we have 2 different settings !? What is the difference?

Storage
![image](https://github.com/mayneyao/eidos/assets/837902/cf14a2ca-a5cc-4bf0-b9ee-84c73a26005c)

Backup
![image](https://github.com/mayneyao/eidos/assets/837902/e0ac65b8-8756-48ed-92b1-096fb5a1ac81)

This icon has no hover text.
![image](https://github.com/mayneyao/eidos/assets/837902/7258e330-f3d2-4025-8480-b92b9df89a0d)

How does the AI part work? What do I select at (1) and how does (2) work? I had set up OpenAI as the provider and supplied my API key. Yet when I select this provider at (2) it does not stay selected as you can see in the screenshot.
![image](https://github.com/mayneyao/eidos/assets/837902/09ab9db3-bbba-4a23-bcad-8aa3981b306a)

**Comment by mayneyao** [comment]
By default, data is stored in OPFS. When you enable NFS, OPFS data is synchronized to local files at regular intervals. The backup tips at the bottom-left corner refer to GitHub backup service. there are different

I'm considering making NFS the default option, which would reduce user confusion (but requires a higher browser version). I also plan to remove the GitHub backup service from the app. This will allow users to decide on their own backup plan, which becomes much simpler when data is stored in a local folder.

You may need to add models in the LLM provider's configuration,  such as `gpt4, gpt-4o`

**Reply by ovizii** [reply]
> By default, data is stored in OPFS. When you enable NFS, OPFS data is synchronized to local files at regular intervals. The backup tips at the bottom-left corner refer to GitHub backup service. there are different

Thanks for clarifying.

> You may need to add models in the LLM provider's configuration, such as `gpt4, gpt-4o`

I have done that already. I also tried lower case: gpt-4o with no change.

![image](https://github.com/mayneyao/eidos/assets/837902/a19b61a1-cadd-48b1-906b-46fb3d6c8c0a)

and yet it just keeps spinning around.

![image](https://github.com/mayneyao/eidos/assets/837902/2db35c10-8774-4fe2-88d7-016eb12319c3)

and even when I select this, it jumps back to "select model"

![image](https://github.com/mayneyao/eidos/assets/837902/fed5337f-8e93-435b-b2a2-accb43cbefff)

**Reply by mayneyao** [reply]
I figured it out, both provider name and model name should be in lowercase 😂

**Comment by mayneyao** [comment]
Currently, lack documentation and guidance, poor onboarding experience. I am working on resolving this issue. Thank you for your suggestion.

**Reply by ovizii** [reply]
No worries, I just wanted to dump my questions somewhere and hopefully inspire you for some FAQ or so :-)

**Reply by mayneyao** [reply]
> No worries, I just wanted to dump my questions somewhere and hopefully inspire you for some FAQ or so :-)

Your feedback is very useful, thank you again.

**Comment by ovizii** [comment]
Any words to add about these two options?

![image](https://github.com/mayneyao/eidos/assets/837902/86364dcd-48a3-45e0-9e48-3d4034521075)

**Comment by ovizii** [comment]
Looks like I need to select “eidosBaseHelper” if I want to enquire about the currently selected document. 

I'm wondering why a “PLAY” symbol is show as an overlay of the “code” response, makes no sense whatsoever. A “COPY” icon would make more sense.

![image](https://github.com/mayneyao/eidos/assets/837902/4351f0e0-22c9-4a56-a7ce-89c46138b2f2)


This is that part of the HTML:

![image](https://github.com/mayneyao/eidos/assets/837902/619d141a-f299-446a-a50c-b0df544c23c8)

**Reply by mayneyao** [reply]
You're right, they provided two different prebuilt prompts. The second one can read the current node's information, markdown for documents and schema for tables.

the Run button is used to manually execute the SQL query when the returned result is in SQL format, so the LLM can know the result. Not all LLM providers support function calling,  I temporarily implemented it this way. It is particularly useful when interacting with table data. for example, "how many records are there in the current table?"

`Use space data` is related to RAG,  but it's still not stable; I should hide it if this experiment is not enabled.

**Reply by ovizii** [reply]
Thanks for confirming. I did indeed enable the RAG experiment, and it seemed to work as you can see. I was just missing the connection to the second prebuilt prompt. 
The "play" thingy is also clear now.

**Comment by ovizii** [comment]
Here are a few more questions, I didn't want to add them myself to your issue about improving the onboarding experience so feel free to pick any that make sense to you.

- [ ] explain nodes
- [ ] improve the available commands to the `command palette` (i.e. I'd like to use keyboard shortcuts instead of using the mouse to lock the current document or hide the sidebar.
- [ ] explain the use/usage of `today `

What exactly is this? I assume it's related to **App: Build your own app using any preferred framework.** but a few more sentences would be great.

![image](https://github.com/mayneyao/eidos/assets/837902/a688cb10-cd8b-46d5-8eb1-ab18be59d7a0)

![image](https://github.com/mayneyao/eidos/assets/837902/213430fb-b558-4230-bf8b-b44d019236d9)

**Reply by mayneyao** [reply]
- ctrl/cmd + `/` toggle chatbot
- ctrl/cmd + `\` toggle sidebar

**Reply by ovizii** [reply]
> * ctrl/cmd + `/` toggle chatbot
> * ctrl/cmd + `\` toggle sidebar

I'm on Win 11 and using Chrome Version 126.0.6478.127 (Official Build) (64-bit) and those two shortcuts do not seem to work for me.
